sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -860,26 +861,52 @@ private void ensureInEventLoop() {
      *
      * @return {@code true} if the given {@link Channel channel} is serviceable; {@code false} otherwise.
      */
-    private boolean isChannelServiceable(final Channel channel, boolean ignoreMaxRequestLimit) {
```

Comment:
why don't we need `ignoreMaxRequestLimit` anymore?

Reply:
Annie borught up a good point that in the fallback-else-path we were incorrectly ignoring maxPendingRequestCount - when reeenforicng it all places passed in false anywhere (except in the high load factor part - where we pick the available channel with the least number of requests (but also should only do it if < maxPendingRequests)

Reply:
Bottom-line - no place should be ignoring maxPendingRequests

Reply:
Closed offline